src/wallet.rs

```diff
@@ -1,82 +1,73 @@
-use crate::bitcoind_rpc::{AddressInfo, Client, Result, Unspent, WalletInfoResponse};
 use bitcoin::{Address, Amount, Transaction, Txid};
+use bitcoincore_rpc::bitcoincore_rpc_json::{
+    AddressType, GetAddressInfoResult, ListUnspentResultEntry, LoadWalletResult,
+};
+use bitcoincore_rpc::RawTx;
+use bitcoincore_rpc::RpcApi;
+use std::result;
 use url::Url;
 
 /// A wrapper to bitcoind wallet
 #[derive(Debug)]
 pub struct Wallet {
     name: String,
-    bitcoind_client: Client,
+    pub bitcoind_client: bitcoincore_rpc::Client,
```

Comment:
But that would mean we will need to write a wrapper for every function we want to use.

Reply:
Yep. I think the client we depend on, whether we roll our own or not, should just be an implementation detail.

Reply:
Mhmm... I don't like it that much as we need to write so much useless code. 

What do you think of removing `Wallet` completely? 😬

Reply:
Haha I wouldn't do it but I understand where you are coming from. Hopefully it's not totally useless if it allows you to simplify calls to the RPC client (by setting optional arguments we don't care about to `None`), use your own types or even combine multiple RPC calls to define higher level functions.

But if it's not doing anything for us, you can just use a newtype, implement `From` and use the actual client. I don't mind.

Reply:
I left it, made it private but provide an explicit function to extract the client if one wants to, e.g. because more access is needed. This allows us to use the lib now, call arbitrary functions from the outside on the client and once identified as useful we can move them into this library and simplify.